### PR TITLE
Writing of RSDS section to EXE with symbols in DBG.

### PR DESCRIPTION
**Pull request by AlexWhiter**

### CODE REVIEW

File: src/PEImage.cpp [hunk 1]
```diff
@@ -331,7 +331,7 @@ bool PEImage::initDbgPtr(bool initDbgDir)
 
 	symtable = (char*)(sec + nsec);
 	nsym = dbg->ExportedNamesSize;
-	strtable = symtable + nsym * IMAGE_SIZEOF_SYMBOL;
+	strtable = symtable + nsym;
```
Comment (rainers) [inline src/PEImage.cpp]: `nsym` is probably wrong to begin with, should be `dbg->ExportedNamesSize/IMAGE_SIZEOF_SYMBOL` and `symtable + dbg->ExportedNamesSize` used here.

File: src/main.cpp [hunk 2]
```diff
@@ -145,13 +145,35 @@ int T_main(int argc, TCHAR* argv[])
...
+	PEImage exe, dbg, *img = NULL;
+	TCHAR dbgname[MAX_PATH];
 
-	CV2PDB cv2pdb(img);
+	if (!exe.loadExe(argv[1]))
+		fatal(SARG ": %s", argv[1], exe.getLastError());
+	if (exe.countCVEntries() || exe.hasDWARF())
+		img = &exe;
+	else
+		if (exe.areDebugSymbolsStripped())
```
Comment (rainers) [inline src/main.cpp]: Is this check necessary? I guess checking the DBG file could always be done.

File: src/main.cpp [hunk 3]
```diff
@@ -145,13 +145,35 @@ int T_main(int argc, TCHAR* argv[])
...
+		{
+			T_strcpy(dbgname, argv[1]);
+			TCHAR *pDot = T_strrchr(dbgname, '.');
+			if (!pDot || pDot <= T_strrchr(dbgname, '/') || pDot <= T_strrchr(dbgname, '\\'))
+				T_strcat(dbgname, TEXT(".dbg"));
+			else
+				T_strcpy(pDot, TEXT(".dbg"));
+
+			if (!dbg.loadExe(dbgname)) 
+				fatal(SARG ": %s", dbgname, dbg.getLastError());
```
Comment (rainers) [inline src/main.cpp]: The most likly cause for failing here is that the DBG file doesn't exist, so the error should still reference the exe file.

File: src/PEImage.cpp (hunk 1 above)
Comment (AlexWhiter) [reply src/PEImage.cpp]: ExportedNames block consist of several null-terminated symbol names with arbitrary length written one after another.
I can send you an example of such file if you need.

File: src/main.cpp (hunk 2 above)
Comment (AlexWhiter) [reply src/main.cpp]: Ok, let's remove this one.

File: src/main.cpp (hunk 3 above)
Comment (AlexWhiter) [reply src/main.cpp]: Another option is a malformed DBG.

File: src/PEImage.cpp (hunk 1 above)
Comment (AlexWhiter) [reply src/PEImage.cpp]: Oops. It seems I've committed wrong version of the file.

**Comment by rainers** [comment]
Thanks for the contribution. I'll do some minor refactoring after merge.

**Comment by rainers** [comment]
> I'll do some minor refactoring after merge.

Done. I hope I did not screw it up.

**Pull request merged**